force-app/main/default/classes/HOT_UpdateSAStatusService.cls

```diff
@@ -0,0 +1,48 @@
+public without sharing class HOT_UpdateSAStatusService {
+    @InvocableVariable
+    public ServiceAppointment serviceAppointment;
+
+    @InvocableMethod
+    public static void updateServiceAppointmentStatus(List<HOT_UpdateSAStatusService> inputVariables) {
+        HOT_UpdateSAStatusService input = inputVariables[0];
+        ServiceAppointment serviceAppointment = input.serviceAppointment;
+        ServiceAppointment serviceAppointmentToUpdate = new ServiceAppointment(Id = serviceAppointment.Id);
+        List<String> errors = new List<String>();
+
+        ServiceAppointment oldServiceAppointment = [
+            SELECT Id, HOT_ServiceResource__r.RelatedRecordId
+            FROM ServiceAppointment
+            WHERE Id = :serviceAppointment.Id
+        ];
+
+        if (oldServiceAppointment.HOT_ServiceResource__r.RelatedRecordId == UserInfo.getUserId()) {
```

Comment:
Skulle du lagt in en else, og kastet en feil om det ikke er samme? Eventuelt bare logge det.